functorch/_src/aot_autograd.py

```diff
@@ -131,29 +136,38 @@ def posthook(grad_input, grad_output):
         )
         node.register_hook(get_posthook(special_stack))
 
+# This class encapsulates all aliasing + mutation info we need about the forward graph
+# See a more detailed overview of the edge case handling at
+# https://docs.google.com/document/d/19UoIh_SVrMy_b2Sx5ZaeOJttm6P0Qmyss2rdBuyfoic/edit
+@dataclass(frozen=True)
+class ViewAndMutationMeta:
+    # length: # forward inputs
+    # For every input, tells us whether the input:
+    # (a) is not mutated
+    # (b) only metadata is mutated
+    # (c) data (and maybe metadta) is mutated
+    mutated_input_info: List[MutationType]
+    # length: (# inputs that have *only* metadata mutations)
+    # We stash the updated FakeTensor that we traced with in the forward in here,
+    # that way we can use it to replay the metadata mutation
+    metadata_mutation_input_info: List[Tensor]
+    # length: # outputs in the compiled forward. Equal to:
+    # length: (# inputs w data mutations) + (# outputs that don't alias inputs)
```

Comment:
meant to put second `length:` here? 

also for readability maybe whitespace after var name before first comment so it's clear that length: goes with the thing below not above?  maybe even put the #length below the description and above the variable so it's sandwiched